src/ln/peer_handler.rs

```diff
@@ -1097,7 +1098,95 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 			}
 		};
 	}
-}
+
+	/// insure we recieved pong message from all peers or disconnect the peers then ping all peers
+	pub fn check_peer(&mut self)-> Result<(), PeerHandleError>{
+			Self::disconnect_if_no_pong(self)?;
+			Self::ping_peers(self)?;
+			Ok(())			
+	}	
+
+	// put a encoded ping message in all peers pending_outbound_buffer
+ 	fn ping_peers(&mut self) -> Result<(), PeerHandleError> {
+
+		for (Descriptor, Peer) in self.peers.lock().unwrap().peers.iter_mut(){
+			// create ping message
+			let ping = msgs::Ping {
+ 				ponglen: 64,
+ 				byteslen: 64
+  			};
+  			// create encoded ping message 
+  			let encoded_ping: &[u8] = &ping.encode();
```

Comment:
You can grep for macro `encode_and_send_msg`, IMO should do the job. No need to comment at this granularity I think

Reply:
I will check out that macro, thanks!
yeah the code speaks clearer than my comments.